pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -280,6 +280,14 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		return reconcile.Result{}, errors.Wrap(err, "failed to create provisioner")
 	}
 
+	provStatus, err := prov.IsReady()
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err,
+			fmt.Sprintf("failed to check services availability"))
+	} else if provStatus.ErrorMessage != "" {
```

Comment:
Elsewhere we use the dirty flag to mean that reconciling needs to be re-enqueued. That seems like it would fit the pattern here, too.

Reply:
Ok, I did not use it since its documentation appeared to be strictly coupled with the host entity, something that here it's not available.